Add doc to explain nesting stories under titles

Glazy opened the pull request:
Issue: N/A

## What I did

Added a section to the 'Writing Stories' documentation explaining how to nest stories under titles in the Storybook sidebar.

Glazy committed:
Add doc to explain nesting stories under titles

shilman reviewed (changes requested):
Thanks @Glazy! Maybe you can also mention that this is configurable using the `hierarchyRootSeparator` option and link to `/configurations/options-parameter` docs?

Glazy committed:
Show user how to change title separator for title

Add a section to the docs to show the user how to change the default
separator ("|") to something else

Glazy commented:
Awesome idea @shilman 💪My latest commit adds a section that explains this :+1:

shilman commented:
Boom!!! 💥
